src/targets/gpu/device/softmax.cpp

```diff
@@ -26,19 +26,19 @@ void softmax(hipStream_t stream, const argument& result, const argument& arg, in
         const std::size_t block_size     = compute_block_size(batch_item_num, max_block_size);
         gs_launch(stream,
                   batch_shape.elements() * block_size,
-                  block_size)([=](auto i, auto idx) __device__ {
+                  block_size)([=] __device__ (auto i, auto idx) {
             auto data_idx = batch.multi(i / block_size);
             using type    = device_type<std::remove_cv_t<typename decltype(input)::value_type>>;
             type init     = lowest();
 
             auto batch_max = block_reduce<max_block_size>(
-                idx, max{}, init, batch_item_num, [&](auto j) __device__ {
+                idx, max{}, init, batch_item_num, [&] __device__ (auto j) {
                     data_idx[axis] = j;
                     return input[data_idx];
                 });
 
             auto batch_sum =
-                block_reduce<max_block_size>(idx, sum{}, 0, batch_item_num, [&](auto j) __device__ {
+                block_reduce<max_block_size>(idx, sum{}, 0, batch_item_num, [&] __device__ (auto j) {
```

Comment:
Attributes for lambdas go after the parameters, specifiers, and `noexcept`.

Reply:
It seems in CUDA/nvcc syntax, you will see warnings if you do that.
```
In file included from /root/AMDMIGraphX/src/targets/gpu/device/argmax.cpp:7:
In file included from /root/AMDMIGraphX/src/targets/gpu/include/migraphx/gpu/device/arg_op.hpp:9:
/root/AMDMIGraphX/src/targets/gpu/device/include/migraphx/gpu/device/reduce.hpp:227:85: warning: nvcc does not allow '__device__' to appear after '()' in lambdas [-Wcuda-compat]
        gs_launch(stream, nelements * block_size, block_size)([=](auto i, auto idx) __device__ {
```

Reply:
Cant we just add `-Wno-cuda-compat` to disable the warning? The issue is that it might break static analyzers that expect C++.